### PR TITLE
Change include semantics

### PR DESCRIPTION
**Pull request by roversch**
As discussed with @tmmsartor , it seems a better idea to have BLASFEO header files include directly what they need, instead of relying on the user of those headers to include the right header files in the right order (e.g. `blasfeo_target.h`).

### CODE REVIEW

**Comment by roversch** [comment]
@giaf I wanted to add you as reviewer, but I don't have those permissions, I think.

**Comment by giaf** [comment]
Good initiative, thanks for that.

I would use the same convention as in acados and HPIPM for the include guard, i.e. BLASFEO_COMMON_H_

And it should be done in a more systematic way, not just on a few files.

**Comment by giaf** [comment]
Why are you disabling the tests in windows?

**Comment by roversch** [comment]
@tmmsartor will look after the include guards

I disabled Windows, because the tests rely on `sys/time.h`, which is not available for MSVC.

File: include/blasfeo_common.h [hunk 1]
```diff
@@ -36,10 +36,14 @@ extern "C" {
 #ifndef BLASFEO_COMMON
 #define BLASFEO_COMMON
 
+#include "blasfeo_target.h"
+
 #ifdef __DSPACE__
 	#define fmax(a, b) (a < b) ? (b) : (a)
 #endif
 
+#include "blasfeo_target.h"
```
Comment (giaf) [inline include/blasfeo_common.h]: why are you including it twice?

File: include/blasfeo_d_aux.h [hunk 2]
```diff
@@ -33,12 +33,18 @@
...
+
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+#ifndef BLASFEO_D_AUX
+#define BLASFEO_D_AUX
 
 
+#include "blasfeo_common.h"
```
Comment (giaf) [inline include/blasfeo_d_aux.h]: again, why are you including the headers twice?

File: include/blasfeo_d_blas.h [hunk 3]
```diff
@@ -26,13 +26,14 @@
...
 
+#include "blasfeo_common.h"
 
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-
+#include "blasfeo_common.h"
```
Comment (giaf) [inline include/blasfeo_d_blas.h]: and here too

**Comment by tmmsartor** [comment]
@roversch  I will fix the Windows `sys/time.h` issue before merging, actually time measurement is not strictly needed in the tests.

**Comment by giaf** [comment]
@tmmsartor thanks, that's great!

File: include/blasfeo_d_blas.h (hunk 3 above)
Comment (roversch) [reply include/blasfeo_d_blas.h]: fixed

File: include/blasfeo_d_aux.h (hunk 2 above)
Comment (roversch) [reply include/blasfeo_d_aux.h]: fixed

File: include/blasfeo_common.h (hunk 1 above)
Comment (roversch) [reply include/blasfeo_common.h]: fixed it

**Comment by giaf** [comment]
I will try to merge later, now I'm a bit busy with Maja...
@tmmsartor your fix with time() fixes tests and probably examples, but not benchmarks, which need to be disabled. Can you implement a BLASFEO timing to fix them all? Thanks!

**Comment by giaf** [comment]
@roversch if you want to help and speed up, blasfeo_common.h is still missing in some single precision headers

**Comment by tmmsartor** [comment]
@giaf, ok I will work on BLASFEO timing.

**Comment by giaf** [comment]
@tmmsartor how far are you from having the blasfeo timings? Should we do it in this PR on in another one?

**Comment by giaf** [comment]
@roversch fmax is in the C99 extension to math.h, so I guess dspace only supports the C89 version of math.h
https://github.com/giaf/blasfeo/issues/51

**Comment by giaf** [comment]
@tmmsartor I guess we keep timings for another PR. Merging now.

**Pull request merged**